app/api/conversations/[conversationId]/messages/utils.ts

```diff
@@ -164,8 +166,22 @@ export async function getRetrievalSystemPrompt(
   rerankEnabled: boolean,
   prioritizeRecent: boolean,
 ) {
-  const response = await getRagieClient().retrievals.retrieve({
-    partition: tenantId,
+  const { ragieApiKey, ragiePartition } = await getTenantRagieSettings(tenantId);
+
+  let client;
+  let partition;
+  if (ragieApiKey) {
+    client = await getTenantRagieClient(ragieApiKey);
+    partition = ragiePartition || undefined;
+  } else {
+    client = getRagieClient();
+    partition = tenantId;
+  }
+
+  assert(!!client, "No client found");
```

Comment:
This seems like it would be cleaner if you could just pass an API key to getRagieClient() which uses the API key if its truthy.  You could decrypt the API in `getTenantRagieSettings`. I would also change that method name to `getRagieSettingsByTenantId()`.